### PR TITLE
chore: inline pupil documentation

### PR DESCRIPTION
**Pull request by garethbowen**
medic/cht-core#9416

# Description

Copies the relevant sections from the pupil readme which is being deleted.
Cleans up a few rendering errors.

medic/cht-core#9416

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: content/en/building/reference/app-settings/_index.md [hunk 1]
```diff
@@ -136,9 +136,27 @@ The following filter functions are available for formatting dates.
...
+
+Non-quoted parameters will be cast to floats (numbers with decimals).
+
+For each validation function, there is also a matching function prepended by `other` that allows you to run functions on other values than the one the rule string is for. This can be useful for fields that have differing requirements depending on another field. For example:
+
+```json
+{
+  "state": "otherEquals('country', 'US') ? lenMin(2) : lenMin(0)"
+}
+```
```
Comment (1yuv) [inline content/en/building/reference/app-settings/_index.md:154]: Can you add th explanation of this check as well.

File: content/en/building/reference/app-settings/_index.md [hunk 2]
```diff
@@ -221,12 +230,9 @@ The following functions are available by default:
 
 ##### Sample usage
 
-For case-insensitive comparison `iEquals` function in Pupil, 
-And you can use `||` for logical OR. Find documentation on these rules in the [Pupil documentation](https://github.com/medic/cht-core/tree/master/shared-libs/validation/src/pupil#rule-strings).
+For case-insensitive comparison `iEquals` function in Pupil, and you can use `||` for logical OR.
```
Comment (1yuv) [inline content/en/building/reference/app-settings/_index.md:233]:
```suggestion
For case-insensitive comparison, use `iEquals` function in Pupil, and you can use `||` for logical OR.
```

**Review by 1yuv — COMMENTED** [review]
Requested explanation of the example.

File: content/en/building/reference/app-settings/_index.md (hunk 1 above)
Comment (garethbowen) [reply content/en/building/reference/app-settings/_index.md:154]: Added. I also realised we don't have any tests for "otherEquals" so I added some.

**Review by 1yuv — APPROVED** [review]

**Pull request merged**